servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGC.java

```diff
@@ -0,0 +1,236 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.jaxrs.gc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.projectnessie.client.NessieConfigConstants.CONF_NESSIE_URI;
+
+import com.google.common.collect.ImmutableMap;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.validation.constraints.NotNull;
+import org.apache.spark.sql.SparkSession;
+import org.projectnessie.api.params.FetchOption;
+import org.projectnessie.client.api.CommitMultipleOperationsBuilder;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.jaxrs.AbstractRest;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.IcebergView;
+import org.projectnessie.model.LogResponse.LogEntry;
+import org.projectnessie.model.Operation;
+import org.projectnessie.model.Operation.Put;
+import org.projectnessie.model.Reference;
+import org.projectnessie.versioned.persist.gc.ContentValues;
+import org.projectnessie.versioned.persist.gc.GCImpl;
+import org.projectnessie.versioned.persist.gc.IdentifiedResult;
+
+public abstract class AbstractRestGC extends AbstractRest {
+
+  @NotNull
+  List<LogEntry> fetchLogEntries(Branch branch, int numCommits) throws NessieNotFoundException {
+    return getApi()
+        .getCommitLog()
+        .refName(branch.getName())
+        .fetch(FetchOption.ALL)
+        .maxRecords(numCommits)
+        .get()
+        .getLogEntries();
+  }
+
+  void fillExpectedContents(Branch branch, int numCommits, Map<String, Set<Content>> expected)
+      throws NessieNotFoundException {
+    fetchLogEntries(branch, numCommits).stream()
+        .map(LogEntry::getOperations)
+        .filter(Objects::nonNull)
+        .flatMap(Collection::stream)
+        .filter(op -> op instanceof Put)
+        .forEach(
+            op -> {
+              Content content = ((Put) op).getContent();
+              expected.computeIfAbsent(content.getId(), k -> new HashSet<>()).add(content);
+            });
+  }
+
+  void performGc(
+      Instant cutoffTimeStamp,
+      Map<String, Instant> cutOffTimeStampPerRef,
+      Map<String, Set<Content>> expectedExpired,
+      Set<String> expectedContentIds,
```

Comment:
Nit: you can make the actual tests more readable by migrating the method signature to
```java 
void performGc(
      Instant cutoffTimeStamp,
      Map<String, Instant> cutOffTimeStampPerRef,
      Map<String, Set<Content>> expectedExpired,
      boolean disableCommitProtection,
      String... expectedContentIds)
```
and construct the `Set` from `expectedContentIds` here. Maybe even more, if you pass the `prefix` into this method as well.

OTOH do you need `expectedContentIds` at all, because those should already be in `expectedExpired`, no?